## Chapter 4 – Exercise 5:  Sử dụng Magic SQL thực hiện lại các bài tập trong phần Chapter 4 – Exericise 2

#### Học viên thực hiện các yêu cầu sau và đối chiếu với kết quả cho trước:

In [1]:
# Câu 1: Liệt kê danh sách các khách hàng ở Brazil, duyệt và xuất ra từng dòng dữ liệu thỏa điều kiện

In [2]:
%load_ext sql

In [3]:
user = 'root'
password = ''
connection_string = "mysql+pymysql://{user}:{password}@localhost:3308/salesdb?charset=UTF8".format(user=user, password=password)
%sql $connection_string

In [4]:
%sql "select * from customers where country='brazil' order by lastname"

 * mysql+pymysql://root:***@localhost:3308/salesdb?charset=UTF8
9 rows affected.


[(15, 'Pedro', 'Afonso', 'Sao Paulo', 'Brazil', '(11) 555-7647'),
 (61, 'Bernardo', 'Batista', 'Rio de Janeiro', 'Brazil', '(21) 555-4252'),
 (62, 'Lúcia', 'Carvalho', 'Sao Paulo', 'Brazil', '(11) 555-1189'),
 (21, 'Aria', 'Cruz', 'Sao Paulo', 'Brazil', '(11) 555-9857'),
 (81, 'Anabela', 'Domingues', 'Sao Paulo', 'Brazil', '(11) 555-2167'),
 (31, 'André', 'Fonseca', 'Campinas', 'Brazil', '(11) 555-9482'),
 (67, 'Janete', 'Limeira', 'Rio de Janeiro', 'Brazil', '(21) 555-3412'),
 (88, 'Paula', 'Parente', 'Resende', 'Brazil', '(14) 555-8122'),
 (34, 'Mario', 'Pontes', 'Rio de Janeiro', 'Brazil', '(21) 555-0091')]

<details>
  <summary>Nhấn vào đây để xem kết quả!</summary>
  <div class="output_subarea output_html rendered_html output_result"><table>
    <tbody><tr>
        <th>Id</th>
        <th>FirstName</th>
        <th>LastName</th>
        <th>City</th>
        <th>Country</th>
        <th>Phone</th>
    </tr>
    <tr>
        <td>15</td>
        <td>Pedro</td>
        <td>Afonso</td>
        <td>Sao Paulo</td>
        <td>Brazil</td>
        <td>(11) 555-7647</td>
    </tr>
    <tr>
        <td>61</td>
        <td>Bernardo</td>
        <td>Batista</td>
        <td>Rio de Janeiro</td>
        <td>Brazil</td>
        <td>(21) 555-4252</td>
    </tr>
    <tr>
        <td>62</td>
        <td>Lúcia</td>
        <td>Carvalho</td>
        <td>Sao Paulo</td>
        <td>Brazil</td>
        <td>(11) 555-1189</td>
    </tr>
    <tr>
        <td>21</td>
        <td>Aria</td>
        <td>Cruz</td>
        <td>Sao Paulo</td>
        <td>Brazil</td>
        <td>(11) 555-9857</td>
    </tr>
    <tr>
        <td>81</td>
        <td>Anabela</td>
        <td>Domingues</td>
        <td>Sao Paulo</td>
        <td>Brazil</td>
        <td>(11) 555-2167</td>
    </tr>
    <tr>
        <td>31</td>
        <td>André</td>
        <td>Fonseca</td>
        <td>Campinas</td>
        <td>Brazil</td>
        <td>(11) 555-9482</td>
    </tr>
    <tr>
        <td>67</td>
        <td>Janete</td>
        <td>Limeira</td>
        <td>Rio de Janeiro</td>
        <td>Brazil</td>
        <td>(21) 555-3412</td>
    </tr>
    <tr>
        <td>88</td>
        <td>Paula</td>
        <td>Parente</td>
        <td>Resende</td>
        <td>Brazil</td>
        <td>(14) 555-8122</td>
    </tr>
    <tr>
        <td>34</td>
        <td>Mario</td>
        <td>Pontes</td>
        <td>Rio de Janeiro</td>
        <td>Brazil</td>
        <td>(21) 555-0091</td>
    </tr>
</tbody></table></div>
  
</details>

In [5]:
# Câu 2: Liệt kê danh sách các nhà cung cấp không có số Fax, duyệt và xuất ra từng dòng dữ liệu thỏa điều kiện
%sql SELECT * FROM suppliers WHERE fax is null

 * mysql+pymysql://root:***@localhost:3308/salesdb?charset=UTF8
16 rows affected.


[(1, 'Exotic Liquids', 'Charlotte Cooper', None, 'London', 'UK', '(171) 555-2222', None),
 (2, 'New Orleans Ca6/Delights', 'Shelley Burke', None, 'New Orleans', 'USA', '(100) 555-4822', None),
 (4, 'Tokyo Traders', 'Yoshi Nagase', None, 'Tokyo', 'Japan', '(03) 3555-5011', None),
 (5, "Cooperativa de Quesos 'Las Cabras'", 'Antonio del Valle Saavedra', None, 'Oviedo', 'Spain', '(98) 598 76 54', None),
 (6, "Mayumi's", 'Mayumi Ohno', None, 'Osaka', 'Japan', '(06) 431-7877', None),
 (8, 'Specialty Biscuits, Ltd.', 'Peter Wilson', None, 'Manchester', 'UK', '(161) 555-4448', None),
 (10, 'Refrescos Americanas LTDA', 'Carlos Diaz', None, 'Sao Paulo', 'Brazil', '(11) 555 4640', None),
 (11, 'Heli Süßwaren GmbH & Co. KG', 'Petra Winkler', None, 'Berlin', 'Germany', '(010) 9984510', None),
 (12, 'Plutzer Lebensmittelgroßmärkte AG', 'Martin Bein', None, 'Frankfurt', 'Germany', '(069) 992755', None),
 (15, 'Norske Meierier', 'Beate Vileid', None, 'Sandvika', 'Norway', '(0) 2-953010', None),
 (16, 'Bigfoot Breweries', 'Cheryl Saylor', None, 'Bend', 'USA', '(503) 555-9931', None),
 (17, 'Svensk Sjöföda AB', 'Michael Björn', None, 'Stockholm', 'Sweden', '08-123 45 67', None),
 (20, 'Leka Trading', 'Chandra Leka', None, 'Singapore', 'Singapore', '555-8787', None),
 (23, 'Karkki Oy', 'Anne Heikkonen', None, 'Lappeenranta', 'Finland', '(953) 10956', None),
 (25, 'Ma Maison', 'Jean-Guy Lauzon', None, 'Montréal', 'Canada', '(514) 555-9022', None),
 (27, 'Escargots Nouveaux', 'Marie Delamare', None, 'Montceau', 'France', '85.57.00.07', None)]

<details>
  <summary>Nhấn vào đây để xem kết quả!</summary>
  <div class="output_subarea output_html rendered_html output_result"><table>
    <tbody><tr>
        <th>Id</th>
        <th>CompanyName</th>
        <th>ContactName</th>
        <th>ContactTitle</th>
        <th>City</th>
        <th>Country</th>
        <th>Phone</th>
        <th>Fax</th>
    </tr>
    <tr>
        <td>1</td>
        <td>Exotic Liquids</td>
        <td>Charlotte Cooper</td>
        <td>None</td>
        <td>London</td>
        <td>UK</td>
        <td>(171) 555-2222</td>
        <td>None</td>
    </tr>
    <tr>
        <td>2</td>
        <td>New Orleans Ca6/Delights</td>
        <td>Shelley Burke</td>
        <td>None</td>
        <td>New Orleans</td>
        <td>USA</td>
        <td>(100) 555-4822</td>
        <td>None</td>
    </tr>
    <tr>
        <td>4</td>
        <td>Tokyo Traders</td>
        <td>Yoshi Nagase</td>
        <td>None</td>
        <td>Tokyo</td>
        <td>Japan</td>
        <td>(03) 3555-5011</td>
        <td>None</td>
    </tr>
    <tr>
        <td>5</td>
        <td>Cooperativa de Quesos 'Las Cabras'</td>
        <td>Antonio del Valle Saavedra</td>
        <td>None</td>
        <td>Oviedo</td>
        <td>Spain</td>
        <td>(98) 598 76 54</td>
        <td>None</td>
    </tr>
    <tr>
        <td>6</td>
        <td>Mayumi's</td>
        <td>Mayumi Ohno</td>
        <td>None</td>
        <td>Osaka</td>
        <td>Japan</td>
        <td>(06) 431-7877</td>
        <td>None</td>
    </tr>
    <tr>
        <td>8</td>
        <td>Specialty Biscuits, Ltd.</td>
        <td>Peter Wilson</td>
        <td>None</td>
        <td>Manchester</td>
        <td>UK</td>
        <td>(161) 555-4448</td>
        <td>None</td>
    </tr>
    <tr>
        <td>10</td>
        <td>Refrescos Americanas LTDA</td>
        <td>Carlos Diaz</td>
        <td>None</td>
        <td>Sao Paulo</td>
        <td>Brazil</td>
        <td>(11) 555 4640</td>
        <td>None</td>
    </tr>
    <tr>
        <td>11</td>
        <td>Heli Süßwaren GmbH &amp; Co. KG</td>
        <td>Petra Winkler</td>
        <td>None</td>
        <td>Berlin</td>
        <td>Germany</td>
        <td>(010) 9984510</td>
        <td>None</td>
    </tr>
    <tr>
        <td>12</td>
        <td>Plutzer Lebensmittelgroßmärkte AG</td>
        <td>Martin Bein</td>
        <td>None</td>
        <td>Frankfurt</td>
        <td>Germany</td>
        <td>(069) 992755</td>
        <td>None</td>
    </tr>
    <tr>
        <td>15</td>
        <td>Norske Meierier</td>
        <td>Beate Vileid</td>
        <td>None</td>
        <td>Sandvika</td>
        <td>Norway</td>
        <td>(0) 2-953010</td>
        <td>None</td>
    </tr>
    <tr>
        <td>16</td>
        <td>Bigfoot Breweries</td>
        <td>Cheryl Saylor</td>
        <td>None</td>
        <td>Bend</td>
        <td>USA</td>
        <td>(503) 555-9931</td>
        <td>None</td>
    </tr>
    <tr>
        <td>17</td>
        <td>Svensk Sjöföda AB</td>
        <td>Michael Björn</td>
        <td>None</td>
        <td>Stockholm</td>
        <td>Sweden</td>
        <td>08-123 45 67</td>
        <td>None</td>
    </tr>
    <tr>
        <td>20</td>
        <td>Leka Trading</td>
        <td>Chandra Leka</td>
        <td>None</td>
        <td>Singapore</td>
        <td>Singapore</td>
        <td>555-8787</td>
        <td>None</td>
    </tr>
    <tr>
        <td>23</td>
        <td>Karkki Oy</td>
        <td>Anne Heikkonen</td>
        <td>None</td>
        <td>Lappeenranta</td>
        <td>Finland</td>
        <td>(953) 10956</td>
        <td>None</td>
    </tr>
    <tr>
        <td>25</td>
        <td>Ma Maison</td>
        <td>Jean-Guy Lauzon</td>
        <td>None</td>
        <td>Montréal</td>
        <td>Canada</td>
        <td>(514) 555-9022</td>
        <td>None</td>
    </tr>
    <tr>
        <td>27</td>
        <td>Escargots Nouveaux</td>
        <td>Marie Delamare</td>
        <td>None</td>
        <td>Montceau</td>
        <td>France</td>
        <td>85.57.00.07</td>
        <td>None</td>
    </tr>
</tbody></table></div>
  
</details>

In [6]:
# Câu 3: Liệt kê danh sách tất cả các sản phẩm có đơn giá từ 30 đến 60 usd
%sql SELECT *, BIN(IsDiscontinued) FROM products WHERE UnitPrice BETWEEN 30 AND 60 ORDER BY UnitPrice DESC

 * mysql+pymysql://root:***@localhost:3308/salesdb?charset=UTF8
20 rows affected.


[(59, 'Raclette Courdavault', 28, Decimal('55.00'), '5 kg pkg.', b'\x00', '0'),
 (51, 'Manjimup Dried Apples', 24, Decimal('53.00'), '50 - 300 g pkgs.', b'\x00', '0'),
 (62, 'Tarte au sucre', 29, Decimal('49.30'), '48 pies', b'\x00', '0'),
 (43, 'Ipoh Coffee', 20, Decimal('46.00'), '16 - 500 g tins', b'\x00', '0'),
 (28, 'Rössle Sauerkraut', 12, Decimal('45.60'), '25 - 825 g cans', b'\x01', '1'),
 (27, 'Schoggi Schokolade', 11, Decimal('43.90'), '100 - 100 g pieces', b'\x00', '0'),
 (63, 'Vegie-spread', 7, Decimal('43.90'), '15 - 625 g jars', b'\x00', '0'),
 (8, 'Northwoods Cranberry Sauce', 3, Decimal('40.00'), '12 - 12 oz jars', b'\x00', '0'),
 (17, 'Alice Mutton', 7, Decimal('39.00'), '20 - 1 kg tins', b'\x01', '1'),
 (12, 'Queso Manchego La Pastora', 5, Decimal('38.00'), '10 - 500 g pkgs.', b'\x00', '0'),
 (56, 'Gnocchi di nonna Alice', 26, Decimal('38.00'), '24 - 250 g pkgs.', b'\x00', '0'),
 (69, 'Gudbrandsdalsost', 15, Decimal('36.00'), '10 kg pkg.', b'\x00', '0'),
 (72, 'Mozzarella di Giovanni', 14, Decimal('34.80'), '24 - 200 g pkgs.', b'\x00', '0'),
 (60, 'Camembert Pierrot', 28, Decimal('34.00'), '15 - 300 g rounds', b'\x00', '0'),
 (64, 'Wimmers gute Semmelknödel', 12, Decimal('33.25'), '20 bags x 4 pieces', b'\x00', '0'),
 (53, 'Perth Pasties', 24, Decimal('32.80'), '48 pieces', b'\x01', '1'),
 (32, 'Mascarpone Fabioli', 14, Decimal('32.00'), '24 - 200 g pkgs.', b'\x00', '0'),
 (26, 'Gumbär Gummibärchen', 11, Decimal('31.23'), '100 - 250 g bags', b'\x00', '0'),
 (10, 'Ikura', 4, Decimal('31.00'), '12 - 200 ml jars', b'\x00', '0'),
 (7, "Uncle Bob's Organic Dried Pears", 3, Decimal('30.00'), '12 - 1 lb pkgs.', b'\x00', '0')]

<details>
  <summary>Nhấn vào đây để xem kết quả!</summary>
  <div class="output_subarea output_html rendered_html output_result"><table>
    <tbody><tr>
        <th>Id</th>
        <th>ProductName</th>
        <th>SupplierId</th>
        <th>UnitPrice</th>
        <th>Package</th>
        <th>IsDiscontinued</th>
        <th>BIN(IsDiscontinued)</th>
    </tr>
    <tr>
        <td>59</td>
        <td>Raclette Courdavault</td>
        <td>28</td>
        <td>55.00</td>
        <td>5 kg pkg.</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>51</td>
        <td>Manjimup Dried Apples</td>
        <td>24</td>
        <td>53.00</td>
        <td>50 - 300 g pkgs.</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>62</td>
        <td>Tarte au sucre</td>
        <td>29</td>
        <td>49.30</td>
        <td>48 pies</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>43</td>
        <td>Ipoh Coffee</td>
        <td>20</td>
        <td>46.00</td>
        <td>16 - 500 g tins</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>28</td>
        <td>Rössle Sauerkraut</td>
        <td>12</td>
        <td>45.60</td>
        <td>25 - 825 g cans</td>
        <td>b'\x01'</td>
        <td>1</td>
    </tr>
    <tr>
        <td>27</td>
        <td>Schoggi Schokolade</td>
        <td>11</td>
        <td>43.90</td>
        <td>100 - 100 g pieces</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>63</td>
        <td>Vegie-spread</td>
        <td>7</td>
        <td>43.90</td>
        <td>15 - 625 g jars</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>8</td>
        <td>Northwoods Cranberry Sauce</td>
        <td>3</td>
        <td>40.00</td>
        <td>12 - 12 oz jars</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>17</td>
        <td>Alice Mutton</td>
        <td>7</td>
        <td>39.00</td>
        <td>20 - 1 kg tins</td>
        <td>b'\x01'</td>
        <td>1</td>
    </tr>
    <tr>
        <td>12</td>
        <td>Queso Manchego La Pastora</td>
        <td>5</td>
        <td>38.00</td>
        <td>10 - 500 g pkgs.</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>56</td>
        <td>Gnocchi di nonna Alice</td>
        <td>26</td>
        <td>38.00</td>
        <td>24 - 250 g pkgs.</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>69</td>
        <td>Gudbrandsdalsost</td>
        <td>15</td>
        <td>36.00</td>
        <td>10 kg pkg.</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>72</td>
        <td>Mozzarella di Giovanni</td>
        <td>14</td>
        <td>34.80</td>
        <td>24 - 200 g pkgs.</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>60</td>
        <td>Camembert Pierrot</td>
        <td>28</td>
        <td>34.00</td>
        <td>15 - 300 g rounds</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>64</td>
        <td>Wimmers gute Semmelknödel</td>
        <td>12</td>
        <td>33.25</td>
        <td>20 bags x 4 pieces</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>53</td>
        <td>Perth Pasties</td>
        <td>24</td>
        <td>32.80</td>
        <td>48 pieces</td>
        <td>b'\x01'</td>
        <td>1</td>
    </tr>
    <tr>
        <td>32</td>
        <td>Mascarpone Fabioli</td>
        <td>14</td>
        <td>32.00</td>
        <td>24 - 200 g pkgs.</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>26</td>
        <td>Gumbär Gummibärchen</td>
        <td>11</td>
        <td>31.23</td>
        <td>100 - 250 g bags</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>10</td>
        <td>Ikura</td>
        <td>4</td>
        <td>31.00</td>
        <td>12 - 200 ml jars</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
    <tr>
        <td>7</td>
        <td>Uncle Bob's Organic Dried Pears</td>
        <td>3</td>
        <td>30.00</td>
        <td>12 - 1 lb pkgs.</td>
        <td>b'\x00'</td>
        <td>0</td>
    </tr>
</tbody></table></div>
  
</details>

In [7]:
# Câu 4: Liệt kê danh sách tất cả các sản phẩm có đơn giá lớn hơn 60 usd và đã ngưng bán
%sql SELECT * FROM products WHERE UnitPrice>60 AND IsDiscontinued

 * mysql+pymysql://root:***@localhost:3308/salesdb?charset=UTF8
2 rows affected.


[(9, 'Mishi Kobe Niku', 4, Decimal('97.00'), '18 - 500 g pkgs.', b'\x01'),
 (29, 'Thüringer Rostbratwurst', 12, Decimal('123.79'), '50 bags x 30 sausgs.', b'\x01')]

<details>
  <summary>Nhấn vào đây để xem kết quả!</summary>
  <div class="output_subarea output_html rendered_html output_result"><table>
    <tbody><tr>
        <th>Id</th>
        <th>ProductName</th>
        <th>SupplierId</th>
        <th>UnitPrice</th>
        <th>Package</th>
        <th>IsDiscontinued</th>
    </tr>
    <tr>
        <td>9</td>
        <td>Mishi Kobe Niku</td>
        <td>4</td>
        <td>97.00</td>
        <td>18 - 500 g pkgs.</td>
        <td>b'\x01'</td>
    </tr>
    <tr>
        <td>29</td>
        <td>Thüringer Rostbratwurst</td>
        <td>12</td>
        <td>123.79</td>
        <td>50 bags x 30 sausgs.</td>
        <td>b'\x01'</td>
    </tr>
</tbody></table></div>
  
</details>

In [8]:
# Câu 5: Liệt kê danh sách các khách hàng trong tên có 2 chữ 'o' (vd: Antonio)
%sql SELECT * FROM `customers` WHERE `LastName` LIKE '%o%o%'

 * mysql+pymysql://root:***@localhost:3308/salesdb?charset=UTF8
6 rows affected.


[(3, 'Antonio', 'Moreno', 'México D.F.', 'Mexico', '(5) 555-3932'),
 (6, 'Hanna', 'Moos', 'Mannheim', 'Germany', '0621-08460'),
 (15, 'Pedro', 'Afonso', 'Sao Paulo', 'Brazil', '(11) 555-7647'),
 (66, 'Maurizio', 'Moroni', 'Reggio Emilia', 'Italy', '0522-556721'),
 (80, 'Miguel', 'Angel Paolino', 'México D.F.', 'Mexico', '(5) 555-2933'),
 (87, 'Pirkko', 'Koskitalo', 'Oulu', 'Finland', '981-443655')]

<details>
  <summary>Nhấn vào đây để xem kết quả!</summary>
  <div class="output_subarea output_html rendered_html output_result"><table>
    <tbody><tr>
        <th>Id</th>
        <th>FirstName</th>
        <th>LastName</th>
        <th>City</th>
        <th>Country</th>
        <th>Phone</th>
    </tr>
    <tr>
        <td>3</td>
        <td>Antonio</td>
        <td>Moreno</td>
        <td>México D.F.</td>
        <td>Mexico</td>
        <td>(5) 555-3932</td>
    </tr>
    <tr>
        <td>6</td>
        <td>Hanna</td>
        <td>Moos</td>
        <td>Mannheim</td>
        <td>Germany</td>
        <td>0621-08460</td>
    </tr>
    <tr>
        <td>15</td>
        <td>Pedro</td>
        <td>Afonso</td>
        <td>Sao Paulo</td>
        <td>Brazil</td>
        <td>(11) 555-7647</td>
    </tr>
    <tr>
        <td>66</td>
        <td>Maurizio</td>
        <td>Moroni</td>
        <td>Reggio Emilia</td>
        <td>Italy</td>
        <td>0522-556721</td>
    </tr>
    <tr>
        <td>80</td>
        <td>Miguel</td>
        <td>Angel Paolino</td>
        <td>México D.F.</td>
        <td>Mexico</td>
        <td>(5) 555-2933</td>
    </tr>
    <tr>
        <td>87</td>
        <td>Pirkko</td>
        <td>Koskitalo</td>
        <td>Oulu</td>
        <td>Finland</td>
        <td>981-443655</td>
    </tr>
</tbody></table></div>
  
</details>

In [9]:
# Câu 6: Liệt kê danh sách các khách hàng có tên bắt đầu bằng chữ 'A' và kết thúc bằng chữ 'E' hoặc ở thành phố bắt đầu bằng 'Ma'
%sql SELECT * FROM `customers` WHERE `LastName` LIKE 'A%o' OR `City` LIKE 'Ma%'

 * mysql+pymysql://root:***@localhost:3308/salesdb?charset=UTF8
7 rows affected.


[(6, 'Hanna', 'Moos', 'Mannheim', 'Germany', '0621-08460'),
 (8, 'Martín', 'Sommer', 'Madrid', 'Spain', '(91) 555 22 82'),
 (9, 'Laurence', 'Lebihan', 'Marseille', 'France', '91.24.45.40'),
 (15, 'Pedro', 'Afonso', 'Sao Paulo', 'Brazil', '(11) 555-7647'),
 (22, 'Diego', 'Roel', 'Madrid', 'Spain', '(91) 555 94 44'),
 (69, 'Alejandra', 'Camino', 'Madrid', 'Spain', '(91) 745 6200'),
 (80, 'Miguel', 'Angel Paolino', 'México D.F.', 'Mexico', '(5) 555-2933')]

<details>
  <summary>Nhấn vào đây để xem kết quả!</summary>
  <div class="output_subarea output_html rendered_html output_result"><table>
    <tbody><tr>
        <th>Id</th>
        <th>FirstName</th>
        <th>LastName</th>
        <th>City</th>
        <th>Country</th>
        <th>Phone</th>
    </tr>
    <tr>
        <td>6</td>
        <td>Hanna</td>
        <td>Moos</td>
        <td>Mannheim</td>
        <td>Germany</td>
        <td>0621-08460</td>
    </tr>
    <tr>
        <td>8</td>
        <td>Martín</td>
        <td>Sommer</td>
        <td>Madrid</td>
        <td>Spain</td>
        <td>(91) 555 22 82</td>
    </tr>
    <tr>
        <td>9</td>
        <td>Laurence</td>
        <td>Lebihan</td>
        <td>Marseille</td>
        <td>France</td>
        <td>91.24.45.40</td>
    </tr>
    <tr>
        <td>15</td>
        <td>Pedro</td>
        <td>Afonso</td>
        <td>Sao Paulo</td>
        <td>Brazil</td>
        <td>(11) 555-7647</td>
    </tr>
    <tr>
        <td>22</td>
        <td>Diego</td>
        <td>Roel</td>
        <td>Madrid</td>
        <td>Spain</td>
        <td>(91) 555 94 44</td>
    </tr>
    <tr>
        <td>69</td>
        <td>Alejandra</td>
        <td>Camino</td>
        <td>Madrid</td>
        <td>Spain</td>
        <td>(91) 745 6200</td>
    </tr>
    <tr>
        <td>80</td>
        <td>Miguel</td>
        <td>Angel Paolino</td>
        <td>México D.F.</td>
        <td>Mexico</td>
        <td>(5) 555-2933</td>
    </tr>
</tbody></table></div>
  
</details>